### DISCUSSION TITLE
Library does not accept my templated member as input for NLOHMANN_DEFINE_TYPE_INTRUSIVE macro

### DISCUSSION THREAD

**Discussion by stormsc1** [nlohmann/json#4280] [Q&A] [answered]
I get the following error on Repository's NLOHMANN_DEFINE_TYPE_INTRUSIVE macro. I can't figure out why?
![image](https://github.com/nlohmann/json/assets/36606323/eee06053-b6fc-4219-803f-aaa77da27093)

Code:
```c++
    template<typename TResource>
    class SoftResourcePtr {
    public:
       ...
        NLOHMANN_DEFINE_TYPE_INTRUSIVE(SoftResourcePtr<TResource>, path)
    private:
        std::filesystem::path path;
        TResource *ptr = nullptr;
    };

    class Repository {
    public:
        NLOHMANN_DEFINE_TYPE_INTRUSIVE(Repository, ..., branches)
    private:
    public:
        ...
        std::vector<SoftResourcePtr<class Branch>> branches;
        ...
    };
```

**Comment by gregmarr** [comment, marked as answer]
That compiles with clang and the latest code just fine.  Can you confirm that there isn't something else in the missing code that is causing the error?  What compiler are you using?  What version of the library?

https://www.godbolt.org/z/6cvTx7j14

**Reply by stormsc1** [reply]
You are right. I left out the constructor for SoftResourcePtr. I was missing a default constructor. How embarrassing. The clang errors helped me out here, msvc was talking black. Thanks <3 :)
